Testscripts/Windows/CPU-Offline.ps1

```diff
@@ -107,41 +114,45 @@ function Main {
 
 		# ##################################################################################
 		# New kernel build for CPU channel change and vmbus interrupt re-assignment
-		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "bash CPUOfflineKernelBuild.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Run-LinuxCmd -ip $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password -command "bash CPUOfflineKernelBuild.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
 		Write-LogInfo "Executing CPUOfflineKernelBuild script inside VM"
 
 		# Wait for kernel compilation completion. 60 min timeout
 		$timeout = New-Timespan -Minutes 60
 		$sw = [diagnostics.stopwatch]::StartNew()
 		while ($sw.elapsed -lt $timeout){
-			$vmCount = $AllVMData.Count
-			Wait-Time -seconds 15
-			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "cat ~/state.txt"
+			Wait-Time -seconds 30
+			$state = Run-LinuxCmd -ip $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password "cat /home/$user/state.txt" -runAsSudo
+			Write-LogDbg "state is $state"
 			if ($state -eq "TestCompleted") {
-				$kernelCompileCompleted = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "cat ~/constants.sh | grep setup_completed=0"
+				$kernelCompileCompleted = Run-LinuxCmd -ip $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password "cat /home/$user/constants.sh | grep setup_completed=0" -runAsSudo
 				if ($kernelCompileCompleted -ne "setup_completed=0") {
-					Write-LogErr "CPUOfflineKernelBuild.sh finished on $($VMData.RoleName) but setup was not successful!"
+					Write-LogErr "CPUOfflineKernelBuild.sh finished on $($AllVMData[0].RoleName) but setup was not successful!"
 				} else {
-					Write-LogInfo "CPUOfflineKernelBuild.sh finished on $($VMData.RoleName)"
-					$vmCount--
+					Write-LogInfo "CPUOfflineKernelBuild.sh finished on $($AllVMData[0].RoleName)"
+					break
 				}
-				break
 			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "CPUOfflineKernelBuild.sh finished with SKIPPED state on $($AllVMData[0].RoleName)!"
 				$resultArr = $resultSkipped
-				throw "CPUOfflineKernelBuild.sh finished with SKIPPED state!"
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
 			} elseif ($state -eq "TestFailed") {
+				Write-LogErr "CPUOfflineKernelBuild.sh finished the failed state on $($AllVMData[0].RoleName)."
```

Comment:
finished with?

Reply:
Fixed in b59736da40b9a97a0de5d6b88cac572431e81fbc